modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java

```diff
@@ -523,6 +523,16 @@ public CodegenOperation fromOperation(String path, String httpMethod, Operation
         Map<String, Schema> definitions = ModelUtils.getSchemas(this.openAPI);
         CodegenOperation op = super.fromOperation(path, httpMethod, operation, servers);
 
+        String pathFormatString = op.path;
+        for (CodegenParameter param : op.pathParams) {
+            // Replace {baseName} with {paramName} for format string
+            String paramSearch = "{" + param.baseName + "}";
+            String paramReplace = "{" + param.paramName + "}";
+
+            pathFormatString = pathFormatString.replace(paramSearch, paramReplace);
+        }
+        op.vendorExtensions.put("pathFormatString", pathFormatString);
```

Comment:
I'll rename this with x-path-format-string to conform the vendor extension naming convention (mentioned in the contributing guidelines)

Reply:
Filed https://github.com/OpenAPITools/openapi-generator/pull/5020